python/py-numpy/Portfile

```diff
@@ -286,38 +72,89 @@ variant."
-                reinplace -E \
-                    "s|_lib_atlas = \\\['atlas'\\\]|_lib_atlas = \\\['tatlas'\\\]|" \
-                    ${worksrcpath}/numpy/distutils/system_info.py
-            } elseif {[file exists ${prefix}/lib/libsatlas.dylib]} {
-                reinplace -E \
-                    "s|_lib_atlas = \\\['atlas'\\\]|_lib_atlas = \\\['satlas'\\\]|" \
-                    ${worksrcpath}/numpy/distutils/system_info.py
-            } else {
-                return -code error "Unable to find Atlas dylibs. Bailing out."
-            }
+    depends_build-append \
+                        path:bin/pkg-config:pkgconfig
+
+    build.env-append    CYTHON=${prefix}/bin/cython-${python.branch}
+
+    # do not allow fallback to (slow) NumPy-internal routines if no external BLAS/LAPACK libraries are found
+    build.args-append -Csetup-args=-Dallow-noblas=false
+
+    compilers.setup     -clang -gcc44 -gcc45 \
+                        -gcc46 -gcc47 -gcc48 -g95
+
+    # https://trac.macports.org/ticket/67136
+    depends_run-append  port:py${python.version}-oldest-supported-numpy
+
+    depends_lib-append  port:fftw-3 \
+                        path:bin/cython-${python.branch}:py${python.version}-cython
+
+    variant accelerate conflicts openblas flexiblas blis description "Use Apples Accelerate Framework" {
+        build.args-append -Csetup-args=-Dblas=accelerate -Csetup-args=-Dlapack=accelerate
+    }
+
+    variant openblas conflicts accelerate flexiblas blis description "Use MacPorts OpenBLAS Libraries" {
+        depends_lib-append  path:lib/libopenblas.dylib:OpenBLAS
+
+        # see: https://trac.macports.org/ticket/66685
+        supported_archs     arm64 ppc ppc64 x86_64
+
+        build.args-append -Csetup-args=-Dblas=openblas -Csetup-args=-Dlapack=openblas \
+                          -Csetup-args=-Dpkg_config_path=${prefix}/lib/pkgconfig
+    }
+
+    variant flexiblas conflicts accelerate openblas blis description "Use MacPorts FlexiBLAS libraries" {
+        depends_lib-append  port:flexiblas
+
+        build.args-append -Csetup-args=-Dblas=flexiblas -Csetup-args=-Dlapack=flexiblas \
+                          -Csetup-args=-Dpkg_config_path=${prefix}/lib/pkgconfig
+    }
+
+    variant blis conflicts accelerate openblas flexiblas description "Use MacPorts BLIS libraries" {
+        depends_lib-append  port:blis
+
+        build.args-append -Csetup-args=-Dblas=blis -Csetup-args=-Dlapack=blis \
+                          -Csetup-args=-Dpkg_config_path=${prefix}/share/pkgconfig
+    }
+
+    # set default BLAS/LAPACK variant if none is explicitly selected by the user
+    if {![variant_isset openblas] && ![variant_isset openblas] && ![variant_isset flexiblas] && ![variant_isset blis]} {
+        # default to macOS Accelerate framework if macOS 11.3 or higher, otherwise OpenBLAS
+        if {${os.platform} eq "darwin" && ({os.major} > 20 || {os.major} ==  20 && ${os.minor} > 2)} {
```

Comment:
Due to a bug in meson, I suggest excluding a case when the compiler is not Clang, since meson cannot find Accelerate with GCC. Otherwise we need to fix meson.

Also, default variants choice seems not to work correctly in the first place.

Reply:
see below